### PR TITLE
Rotation support.

### PR DESCRIPTION
**Pull request by rafael-chavez**
## Change list

Please provide briefly described change list which are you going to propose. 
 
## Types of changes

What types of changes are you proposing/introducing to Java client?
_Put an `x` in the boxes that apply_

- [ ] Bugfix (non-breaking change which fixes an issue)
- [X] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## Details
* Rotation support to fit mobile-spec found here https://github.com/SeleniumHQ/mobile-spec/blob/master/spec-draft.md#device-rotation *

* Dependency changes need to be changed to work with latest source for selenium.*

Please provide more details about changes if it is necessary. If there are new features you can provide code samples which show the way they
work and possible use cases. Also you can create [gists](https://gist.github.com) with pasted java code samples or put them here using markdown. 
About markdown please read [Mastering markdown](https://guides.github.com/features/mastering-markdown/) and [Writing on GitHub](https://help.github.com/categories/writing-on-github/)

### CODE REVIEW

**Comment by rafael-chavez** [comment]
@TikhomirovSergey

File: src/main/java/io/appium/java_client/AppiumDriver.java [hunk 1]
```diff
@@ -630,7 +631,24 @@ protected void setSetting(AppiumSetting setting, Object value) {
...
+    
+    @Override
+    public void rotate(DeviceRotation rotation) {
+        execute(DriverCommand.SET_SCREEN_ROTATION,
+            ImmutableMap.of("orientation", rotation.parameters()));
+    }
+    
+    @Override
+    public DeviceRotation rotation() {
+        Response response = execute(DriverCommand.GET_SCREEN_ROTATION);
```
Comment (rafael-chavez) [inline src/main/java/io/appium/java_client/AppiumDriver.java]: ah sorry let me update this, forgot to commit this part.

File: pom.xml [hunk 2]
```diff
@@ -29,14 +29,28 @@
...
+        </dependency>
+        <dependency>
+            <groupId>org.seleniumhq.selenium</groupId>
+            <artifactId>selenium-chrome-driver</artifactId>
+            <version>2.53.0</version>
+        </dependency>
+        <dependency>
+            <groupId>org.seleniumhq.selenium</groupId>
             <artifactId>selenium-java</artifactId>
             <version>2.53.0</version>
```
Comment (TikhomirovSergey) [inline pom.xml]: Why did you remove this?

**Comment by TikhomirovSergey** [comment]
Hi @rafael-chavez 
I have found the PR that you proposed to Selenium
https://ecosystem.atlassian.net/wiki/display/SELENIUM/Building+Page+Objects+with+Atlassian+Selenium

**Comment by TikhomirovSergey** [comment]
Ok, Firstly
Could you improve the issue which has been found by codecy?

File: pom.xml [hunk 3]
```diff
@@ -29,14 +29,28 @@
...
-            <exclusions>
```
Comment (TikhomirovSergey) [inline pom.xml]: Why did you remove this?

**Comment by TikhomirovSergey** [comment]
It is interesting. 
I think it is great to have a feature like that. But there are problems

- The Selenium Java clint bundle still doesn't contain the DeviceRotation  class.

- are you sure that Appium server supports [this command](https://github.com/SeleniumHQ/selenium/pull/2474/files#diff-578329ec7273c4a024979fd0ebcda293R187)? Please take a look at [that string](https://github.com/appium/appium/blob/81d2e9cb74dccfa49a57a9539a0c22999235c9c9/lib/server/routing.js#L111). @imurchie @jlipps Could you claryfy what does this command do?

File: src/main/java/io/appium/java_client/AppiumDriver.java (hunk 1 above)
Comment (rafael-chavez) [inline src/main/java/io/appium/java_client/AppiumDriver.java]: The variable is being used.

**Comment by SrinivasanTarget** [comment]
It's really a cool feature 👍  I see we have this endpoint implemented only in ```appium-ios-driver``` here https://github.com/appium/appium-ios-driver/blob/master/lib/commands/gesture.js#L63

**Comment by rafael-chavez** [comment]
@SrinivasanTarget actually the link you pasted is for the endpoint /rotate 

the [mobile spec](https://github.com/SeleniumHQ/mobile-spec/blob/master/spec-draft.md#device-rotation) defines a /rotation endpoint. Maybe a good idea to merge these two? im not sure what spec this mobileRotation command was adhering to. Maybe @jlipps @imurchie can give more details?

**Comment by rafael-chavez** [comment]
@TikhomirovSergey DeviceRotation class is proposed to be part of selenium project.
https://github.com/SeleniumHQ/selenium/pull/2482

**Comment by rafael-chavez** [comment]
@SrinivasanTarget i updated the routes in the appium server to support this [command](https://github.com/appium/appium-base-driver/pull/67). The [selenium](https://github.com/SeleniumHQ/selenium/pull/2482) has been merged already.

**Comment by TikhomirovSergey** [comment]
@rafael-chavez Ok. It will be merged when it will be available it selenium. Are you agree?

**Comment by rafael-chavez** [comment]
👍

**Comment by TikhomirovSergey** [comment]
Yes

**Comment by SrinivasanTarget** [comment]
Closing this in favor of #489
